### PR TITLE
add getBucketByIndex API and ioctl commands to get vote bucket data

### PR DESCRIPTION
**Pull request by koseoyoung**
works on #2162 
add ```ReadStakingDataMethod_BUCKET_BY_INDEX ``` method and bucket/bucketlist ioctl commands

### CODE REVIEW

File: ioctl/cmd/bc/bucket.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	if output.Format == "" {
+		message := fmt.Sprintf("Blockchain Node: %s\n%s", m.Node, output.JSONString(m.Bucket))
+		return message
+	}
+	return output.FormatString(output.Result, m)
+}
+
+// getBucket get bucket from chain
+func getBucket(arg string) error {
+	bucketindex, err := strconv.ParseUint(arg, 10, 64)
```
Comment (dustinxie) [inline ioctl/cmd/bc/bucket.go:66]: check this err?

File: ioctl/cmd/bc/bcblock.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ var (
 
 // bcBlockCmd represents the bc Block command
 var bcBlockCmd = &cobra.Command{
-	Use:   "block [HEIGHT|HASH]",
+	Use:   config.TranslateInLang(bcBlockCmdUses, config.UILanguage),
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcblock.go:41]: good catch, thx

File: ioctl/cmd/bc/bcbucketlist.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/output"
+	"github.com/iotexproject/iotex-core/ioctl/util"
+)
+
+// Multi-language support
+var (
+	bcBucketListCmdShorts = map[config.Language]string{
+		config.English: "Get bucket list from block chain with given address",
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:30]: "Get bucket list for given address on IoTeX blockchain",

File: ioctl/cmd/bc/bcbucketlist.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		config.Chinese: "",
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:31]: "在IoTeX区块链上读取账户地址的投票列表"

File: ioctl/cmd/bc/bcbucketlist.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	}
+	bcBucketListUses = map[config.Language]string{
+		config.English: "bucketlist [ALIAS|ADDRESS]",
+		config.Chinese: "",
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:35]: bucketlist [别名|地址]

File: ioctl/cmd/bc/bucket.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/output"
+	"github.com/iotexproject/iotex-core/ioctl/util"
+)
+
+// Multi-language support
+var (
+	bcBucketCmdShorts = map[config.Language]string{
+		config.English: "Get bucket from block chain with given bucket index",
```
Comment (dustinxie) [inline ioctl/cmd/bc/bucket.go:30]: "Get bucket for given index on IoTeX blockchain",

File: ioctl/cmd/bc/bucket.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+		config.Chinese: "",
```
Comment (dustinxie) [inline ioctl/cmd/bc/bucket.go:31]: "在IoTeX区块链上根据索引读取投票"

File: ioctl/cmd/bc/bucket.go [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+	}
+	bcBucketUses = map[config.Language]string{
+		config.English: "bucket [BUCKET_INDEX]",
+		config.Chinese: "",
```
Comment (dustinxie) [inline ioctl/cmd/bc/bucket.go:35]: "bucket [票索引]"

File: ioctl/cmd/bc/bcbucketlist.go [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+func getBucketList(arg string) error {
+	address, err := util.GetAddress(arg)
+	if err != nil {
+		return output.NewError(output.AddressError, "", err)
+	}
+	bucketlist, err := getBucketListByAddress(address)
+	if err != nil {
+		return err
+	}
+	message := bucketlistMessage{Node: config.ReadConfig.Endpoint, Bucketlist: bucketlist}
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:82]: can you paste an example output?

File: ioctl/cmd/bc/bcbucketlist.go [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+	fmt.Println(message.String())
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:83]: can you paste an example output?

File: ioctl/cmd/bc/bucket.go [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+// getBucket get bucket from chain
+func getBucket(arg string) error {
+	bucketindex, err := strconv.ParseUint(arg, 10, 64)
+	bucket, err := getBucketByIndex(bucketindex)
+	if err != nil {
+		return err
+	}
+	message := bucketMessage{Node: config.ReadConfig.Endpoint, Bucket: bucket}
+	fmt.Println(message.String())
```
Comment (dustinxie) [inline ioctl/cmd/bc/bucket.go:72]: can you paste an example output?

File: ioctl/cmd/bc/bcbucketlist.go (hunk 10 above)
Comment (koseoyoung) [reply ioctl/cmd/bc/bcbucketlist.go:83]:
```Blockchain Node: api.testnet.iotex.one:443
{
  "index": 34,
  "candidateAddress": "io1t56twy23yjuqscljpjc869hyqw3gpswwj0g228",
  "stakedAmount": "100000000000000000000",
  "stakedDuration": 7,
  "createTime": {
    "seconds": 1589179280
  },
  "stakeStartTime": {
    "seconds": 1589179280
  },
  "unstakeStartTime": {},
  "owner": "io120au9ra0nffdle04jx2g5gccn6gq8qd4fy03l4"
}
{
  "index": 43,
  "candidateAddress": "io1l40dc4q95fsjdprhcga5qrpdsj46q3wpzr27n9",
  "stakedAmount": "170000000000000000000",
  "stakedDuration": 14,
  "createTime": {
    "seconds": 1589424135
  },
  "stakeStartTime": {
    "seconds": 1589424135
  },
  "unstakeStartTime": {},
  "owner": "io120au9ra0nffdle04jx2g5gccn6gq8qd4fy03l4"
}

File: ioctl/cmd/bc/bucket.go (hunk 11 above)
Comment (koseoyoung) [reply ioctl/cmd/bc/bucket.go:72]: we didn't deploy for this API yet, so I can't see the output yet

File: ioctl/cmd/bc/bcbucketlist.go (hunk 10 above)
Comment (dustinxie) [reply ioctl/cmd/bc/bcbucketlist.go:83]: 1. "stakedAmount": 100 IOTX
2. "stakeStartTime": use time.Format() to output time in string format
3. "stakedDuration": 7 days,
4. move "owner" to after "index"
5. "candidateAddress" --> "candidate"

Comment (koseoyoung) [reply ioctl/cmd/bc/bcbucketlist.go:122]:
```
Blockchain Node: api.testnet.iotex.one:443
{
index: 34
owner: io120au9ra0nffdle04jx2g5gccn6gq8qd4fy03l4
candidate: io1t56twy23yjuqscljpjc869hyqw3gpswwj0g228
stakedAmount: 100 IOTX
stakedDuration: 7 days
createTime: 2020-05-11T06:41:20Z
stakeStartTime: 2020-05-11T06:41:20Z
unstakeStartTime: none
}
{
index: 43
owner: io120au9ra0nffdle04jx2g5gccn6gq8qd4fy03l4
candidate: io1l40dc4q95fsjdprhcga5qrpdsj46q3wpzr27n9
stakedAmount: 170 IOTX
stakedDuration: 14 days
createTime: 2020-05-14T02:42:15Z
stakeStartTime: 2020-05-14T02:42:15Z
unstakeStartTime: none
}

File: ioctl/cmd/bc/bcbucketlist.go [hunk 12]
```diff
@@ -75,11 +78,47 @@ func getBucketList(arg string) error {
...
+		unstakeStartTime, err := ptypes.Timestamp(b.UnstakeStartTime)		
+		if err != nil {
+			return output.NewError(output.ConvertError, "failed to convert to time", err)
+		}
+		unstakeStartTimeFormat := "none"
+		if unstakeStartTime != time.Unix(0, 0).UTC() {
+			unstakeStartTimeFormat = unstakeStartTime.Format(time.RFC3339)
+		} 
+		bucketlist = append(bucketlist, &bucket{
+			index: b.Index,
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:108]: make the above code newBucket()
append(bucketlist, newBucket(b)

File: ioctl/cmd/bc/bucket.go [hunk 13]
```diff
@@ -67,11 +97,43 @@ func getBucket(arg string) error {
...
+	if err != nil {
+		return output.NewError(output.ConvertError, "failed to convert to time", err)
+	}
+	unstakeStartTimeFormat := "none"
+	if unstakeStartTime != time.Unix(0, 0).UTC() {
+		unstakeStartTimeFormat = unstakeStartTime.Format(time.RFC3339)
+	} 
+	message := bucketMessage{ 
+		Node: config.ReadConfig.Endpoint, 
+		Bucket: &bucket {
```
Comment (dustinxie) [inline ioctl/cmd/bc/bucket.go:126]: use newBucket(b) same as above

File: ioctl/cmd/bc/bcbucketlist.go [hunk 14]
```diff
@@ -75,11 +78,47 @@ func getBucketList(arg string) error {
...
+		if err != nil {
+			return output.NewError(output.ConvertError, "failed to convert to time", err)
+		}
+		unstakeStartTime, err := ptypes.Timestamp(b.UnstakeStartTime)		
+		if err != nil {
+			return output.NewError(output.ConvertError, "failed to convert to time", err)
+		}
+		unstakeStartTimeFormat := "none"
+		if unstakeStartTime != time.Unix(0, 0).UTC() {
+			unstakeStartTimeFormat = unstakeStartTime.Format(time.RFC3339)
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcbucketlist.go:105]: can we also use ptypes.Timestamp, so no need for Format(time.RFC3339)

File: action/protocol/staking/read_state.go [hunk 15]
```diff
@@ -79,6 +79,16 @@ func readStateBucketsByCandidate(ctx context.Context, sr protocol.StateReader, c
 	return toIoTeXTypesVoteBucketList(buckets)
 }
 
+func readStateBucketByIndex(ctx context.Context, sr protocol.StateReader,
```
Comment (yutongp) [inline action/protocol/staking/read_state.go:82]: not sure what's the major use case for this.
if we expect user has a list of indexes to query. a `bucketsByIndexes` is better which reduce number of api calls

**Pull request closed without merging**